### ISSUE TITLE
Side effects between shinyFiles dialogs and shiny::renderUI elements 

### ISSUE THREAD

**Issue by akaever** [thomasp85/shinyFiles#23]
There seems to be an unwanted side effect between shinyFiles dialogs and the shiny functions renderUI and uiOutput. When opening a shinyFiles dialog, UI elements generated by uiOutput are reset to their initial value. UI elements generated directly in the shiny UI part are not affected. Tested with sliderInput and textInput, see

``` R
library(shiny)
library(shinyFiles)

runApp(list(
    ui = bootstrapPage(
      uiOutput("testSlider1"), br(),
      sliderInput("testSliderInput2", "testSliderInput2", 0, 100, 1, step = 1), br(),

      uiOutput("testText1"), br(),
      textInput("testTextInput2", "testTextInput2", value = ""), br(), 

      shinyFilesButton("testFilesButton", label="testFilesButton", title="testFilesButton", multiple=FALSE)
    ),
    server = function(input, output, session) {
      output$testSlider1 <- renderUI({
        sliderInput("testSliderInput1", "testSliderInput1", 0, 100, 1, step = 1)
      })

      output$testText1 <- renderUI({
        textInput("testTextInput1", "testTextInput1", value = "")
      })

      shinyFileChoose(input, "testFilesButton", session=session, roots=c(wd='.'))
    }
))
```

Thanks

**Comment by thomasp85** [comment]
Thanks for the report - I've been off with a concussion, hence the late reply.

My feeling is that this is due to Shiny as similar cases have been. Thankfully they have been quite good at responding to these things so they probably will be in this case too - I'll look into it...

**Comment by thomasp85** [comment]
This is indeed a Shiny bug residing in the unbindAll-bindAll pattern - I'll notify the Shiny devs...

**Comment by burgerga** [comment]
Link to Shiny bug: https://github.com/rstudio/shiny/issues/1244

**Comment by mathematiguy** [comment]
Is this still a problem? I seem to be having the same issue with a renderUI form containing a shinyFilesButton where all of the form fields are reset on click.

When I implement the ui not inside a renderUI element, I can show that the same UI code works fine.

I happen to be using shiny_1.0.1, shinyFiles_0.6.2 and R version 3.3.3 (2017-03-06).

**Comment by thomasp85** [comment]
This is still an issue, yes.

**Comment by nathmenini** [comment]
Did they fix it? Or is this still a problem?

**Comment by vnijs** [comment]
Looking at @akaever's example, simply removing all calls to `Shiny.unbindAll();` and `Shiny.bindAll();` in shinyFiles.js seems to fix the problem. Everything seems to work as intended after this change but I assume this will cause other problems however.

According to this [link](https://stackoverflow.com/questions/36340402/limit-shiny-unbindall-and-bindall-only-to-and-html-element) you can specify a specific element to bind / unbind. It is not clear what elements specifically would need to be (un)bound and how to specify them.

**Comment by thomasp85** [comment]
Closed by 1c82b5e4c67565afc740cf925c9eb98962db7e6d
